docs/revive_compiler/differences_yul_translation.md

```diff
@@ -0,0 +1,98 @@
+In general, revive preserves the memory layout, meaning low level memory operations are supported. However, a few caveats apply:
+- The EVM linear heap memory is emulated using a fixed byte buffer of 64kb. This implies that the maximum memory a contract can use is limited to 64kbit (on Ethereum, contract memory is capped by gas and therefor variable).
+- Thus, accessing memory offsets larger than the fixed buffer size will trap the contract at runtime with `OutOfBound` erros.
+- The compiler might detect and optimize unused memory reads and writes, leading to a different `msize` compared to what the EVM would see. 
+
+### `calldataload`, `calldatacopy`
+In the constructor code, the offset is ignored and this always returns `0`.
+
+### `codecopy`
+Only supported in constructor code.
+
+### `invalid`
+Traps the contract but does not consume the remaining gas.
+
+### `call`, `delegatecall`, `staticall`
+Calls ignore the supplied gas limit (if any) but forward all remaining resources.
+The reason is multifold, see [Differences to Ethereum](../differences_to_eth.md) (section `Cross-Contract Call Gas Limit`).
+
+Contract authors are strongly advised to [protect against re-entrancy attacks](https://docs.soliditylang.org/en/latest/security-considerations.html#reentrancy).
+
+:::warning
+
+The `solc` compiler supplies a gas stipend of `2300` to calls resulting from `address payable.{send,transfer}`. Which means: The resource limits solc injects for balance transfer calls will be ignored and not protect against re-entrancy attacks.
+
+`revive` uses a heuristic to detect this. If detected, the compiler will disable call re-entrancy to protect against re-entrancy attacks with balance transfers (child context resources still remain uncapped). But this is just a heuristc and not guaranteed behavior.
+
+Note: Using `address payable.{send,transfer}` [is considered deprecated since a long time ago](https://diligence.consensys.io/blog/2019/09/stop-using-soliditys-transfer-now/) anyways. There is a revive pre-compile planned for making safe balance transfers.
+
+:::
+
+### `create`, `create2`
+Deployments on revive work different than on EVM, see also [Differences to Ethereum](../differences_to_eth.md) (section `PolkaVM instead of EVM`). In a nutshell: Instead of supplying the deploy code concatenated with the constructor arguments (the EVM deploy model), the [revive runtime expects two pointers](https://docs.rs/pallet-revive/latest/pallet_revive/trait.SyscallDoc.html#tymethod.instantiate):
+1. A buffer containing the code hash to deploy.
+2. The constructor arguments buffer.
+
+To make the `new` keyword in Solidity work seamlessly, `revive` translates the `dataoffset` and `datasize` instructions so that they assume the contract hash instead of the contract code. The hash is always of constant size. Thus, `revive` is able to supply the expected code hash and constructor arguments pointer to the runtime. 
+
```

Comment:
maybe a sample solidity example here would help

Reply:
I hope Solidity devs reading this would know about contract instantiation :see_no_evil:  Or do you mean a broken example?